base/dict.jl

```diff
@@ -8,7 +8,10 @@ const secret_table_token = :__c782dbf1cf4d6a2e5e3865d7e95634f2e09b5902__
 
 haskey(d::Associative, k) = in(k,keys(d))
 
-function in(p::Tuple{Any,Any}, a::Associative)
+function in(p, a::Associative)
+    if !isa(p,Pair)
+        error("Associative collections only contain Pairs; look for e.g. A=>B instead")
```

Comment:
@JeffBezanson I thought that `function in(p, a::Associative)` and `function in(p::Pair, a::Associative)` would fallback correctly/equivalently? (or specific error for `function in(p::Tuple{Any,Any}, a::Associative)`).

+1 to directing towards `A in keys(a)` and `B in values(a)`